File: csharp/bob/Bob.cs
using System;

public static class Bob
{
    public static string Response(string statement)
    {
        statement = statement.Trim();
        if (string.IsNullOrEmpty(statement))
        {
            return "Fine. Be that way!";
        }

        if (statement.ToLower() != statement && statement.ToUpper() == statement)
        {
            if (statement.EndsWith('?'))
            {
                return "Calm down, I know what I'm doing!";
            }

            return "Whoa, chill out!";
        }

        if (statement.EndsWith('?'))
        {
            return "Sure.";
        }

        return "Whatever.";
    }
}

File: elixir/bob/lib/bob.ex
defmodule Bob do
  defp empty?(input), do: String.trim(input) == ""
  defp letters?(input), do: String.match?(input, ~r/[[:alpha:]]/)
  defp question?(input), do: String.trim(input) |> String.ends_with?("?")
  defp yelling?(input), do: input == String.upcase(input) && letters?(input)
  defp yelling_question?(input), do: yelling?(input) && question?(input)

  def hey(input) when empty?(input) do
    "Fine. Be that way!"
  end

  def hey(input) do
    cond do
      yelling_question?(input) ->
        "Calm down, I know what I'm doing!"

      yelling?(input) ->
        "Whoa, chill out!"

      question?(input) ->
        "Sure."

      "Anything else?" ->
        "Whatever."
    end
  end
end
